cmd2/cmd2.py

```diff
@@ -448,6 +436,37 @@ def __init__(self, completekey: str='tab', stdin=None, stdout=None, persistent_h
         # If this string is non-empty, then this warning message will print if a broken pipe error occurs while printing
         self.broken_pipe_warning = ''
 
+        # Check if history should persist
+        if persistent_history_file and rl_type != RlType.NONE:
+            persistent_history_file = os.path.expanduser(persistent_history_file)
+            read_err = False
+
+            try:
+                # First try to read any existing history file
+                readline.read_history_file(persistent_history_file)
+            except FileNotFoundError:
+                pass
+            except OSError as ex:
+                self.perror("readline cannot read persistent history file '{}': {}".format(persistent_history_file, ex),
+                            traceback_war=False)
+                read_err = True
+
+            if not read_err:
+                try:
+                    # Make sure readline is able to write the history file. Doing it this way is a more thorough check
+                    # than trying to open the file with write access since readline's underlying function needs to
+                    # create a temporary file in the same directory and may not have permission.
+                    readline.set_history_length(persistent_history_length)
+                    readline.write_history_file(persistent_history_file)
```

Comment:
Thanks for adding better error checking to make sure the user can actually write to the persistent history file.